helm/v3: upgrade to Helm v3 stable

hiddeco opened the pull request:
Fixes #91

hiddeco committed:
release: declare `AntecedentAnnotation`

This commit drops the dependency on the `cluster/kubernetes` package
from Flux by moving the `AntecedentAnnotation` to our own code base.

The main reason to do this now is because we want to be able to define
our own Kubernetes package versions, something that is not possible
without this change due to `cluster/kubernetes` (indirectly) depending
on various `k8s.io/*` packages.

hiddeco committed:
helm/v3: upgrade to Helm v3 stable

This includes a small refactor of the Kubernetes configuration setup,
so that we no longer depend on any `k8s.io/kubernetes/*` packages.
It also includes a bump of our `k8s.io/*` packages from
`kubernetes-1.14.4` to `kubernetes-1.16.2`, as required by Helm 3.

hiddeco committed:
build: enforce k8s.io/client-go upgrade

2opremio commented:
I don't fully understand the upgrade and downgrade replacement dance. Can you please clarify it? (Ideally in comments) and indicate what's the way forward to get rid of them?

stefanprodan reviewed (approved):
Awesome! Lets get this merged and update the hands-on

hiddeco commented:
> I don't fully understand the upgrade and downgrade replacement dance.

`go mod tidy` reaches the conclusion that we are requiring `v11.0.0` due to downstream dependencies. This is however not true, as we require the version that equals to the `kubernetes-1.16.2` tag, the replace enforces this so that Go Modules uses the right version. Otherwise `make` results in:

```console
go install ./cmd/helm-operator
# k8s.io/client-go/rest
../../go/pkg/mod/k8s.io/client-go@v11.0.0+incompatible/rest/request.go:598:31: not enough arguments in call to watch.NewStreamWatcher
	have (*versioned.Decoder)
	want (watch.Decoder, watch.Reporter)
make: *** [Makefile:114: /home/hidde/go/bin/bin/helm-operator] Error 2
```

We depend on Flux because of the `resource`, `checkpoint` and `git` packages it offers.

The overwrite for `k8s.io/code-generator` can actually be removed, as this is now managed outside the application's dependencies in `hack/tools/go.mod`.

hiddeco committed:
build: remove replace for `k8s.io/code-generator`

The dependency of the code-generator version is managed outside of
the application's dependencies in `hack/tools/go.mod`.
